src/jti_content.js

```diff
@@ -1,169 +1,244 @@
-                        descriptionElement.addEventListener('change', descriptionChangeEvent);
-                    }
-                }
-            });
-        }
-    }
-}
-
-// Create observer to monitor for description field
-var observer = new MutationObserver(function (mutations) {
-    mutations.forEach(observeDocumentBody);
-});
-observer.observe(document.body, {subtree: true, attributes: true, attributeFilter: ['resolved']});
+/* Copyright 2016 Redbrick Technologies, Inc. */
+/* https://github.com/rdbrck/jira-description-extension/blob/master/LICENSE */
+
+/* global chrome, browser */
+
+var StorageID = 'Jira-Template-Injector';
+
+var browserType = 'Chrome'; // eslint-disable-line no-unused-vars
+if (navigator.userAgent.indexOf('Firefox') !== -1 || navigator.userAgent.indexOf('Edge') !== -1) {
+    chrome = browser; // eslint-disable-line no-native-reassign
+    chrome.storage.sync = browser.storage.local;
+    if (navigator.userAgent.indexOf('Firefox') !== -1) {
+        browserType = 'Firefox';
+    }
+    if (navigator.userAgent.indexOf('Edge') !== -1) {
+        browserType = 'Edge';
+    }
+}
+
+// Handle <TI> tag selection.
+$(document).on('click', '#description', function () {
+    var text = $(this).val(),
+        ctrlDown = false,
+        ctrlKey = 17,
+        cmdKey = 91,
+        backtickKey = 192,
+        cursorStart = $(this).prop('selectionStart'),
+        cursorFinish = $(this).prop('selectionEnd'),
+        end = (text.length - 5),
+        selectStart = null,
+        selectEnd = null,
+        i = 0;
+
+    // Only proceed if this is a click. i.e. not a highlight.
+    if (cursorStart === cursorFinish) {
+        // Look for opening tag '<TI>'.
+        for (i = cursorStart; i >= 4; i--) {
+            if (i !== 4) {
+                if (text.slice((i - 5), i) === '</TI>') {
+                    // Found closing tag before opening tag -> We are not withing any valid tags.
+                    break;
+                }
+            }
+            if (text.slice((i - 4), i) === '<TI>') {
+                // Found opening Tag!
+                selectStart = (i - 4);
+                break;
+            }
+        }
+
+        if (selectStart) {
+            // Look for closing tag '</TI>'
+            for (i = cursorStart; i <= end; i++) {
+                if (text.slice(i, (i + 4)) === '<TI>') {
+                    // Found another opening bracket before closing bracket. Exit search.
+                    break;
+                }
+                if (text.slice(i, (i + 5)) === '</TI>') {
+                    // Found closing Tag!
+                    selectEnd = (i + 5);
+                    break;
+                }
+            }
+            if (selectEnd) {
+                // Select all the text between the two tags.
+                $(this)[0].setSelectionRange(selectStart, selectEnd);
+            }
+        }
+    }
+
+    // detect ctrl or cmd pressed

```

Comment:
I like to start comments with a capital -> Detect.
Also I think we should be able to determine which os we are on and from there only detect one or the other. 
if mac detect cmd
else detect ctrl